File: 4363-minimum-total-price-after-applying-discounts/minimum-total-price-after-applying-discounts.py
class Solution(object):
    def minPrice(self, prices, discounts):
        prices.sort(reverse = True)
        discounts.sort(reverse = True)
        total = j = 0
        for i in prices:
            if(j < len(discounts)):
                total += (i * (100 - discounts[j]) / 100.0)
                j += 1
            else :
                total += i
        return total

File: 4363-minimum-total-price-after-applying-discounts/minimum-total-price-after-applying-discounts.cpp
class Solution {
public:
    double minPrice(vector<int>& prices, vector<int>& discounts) {
        sort(prices.rbegin(), prices.rend());
        sort(discounts.rbegin() , discounts.rend());
        double total = 0 , j = 0;
        for(int i:prices){
            if(j < discounts.size()) total += (i * (100 - discounts[j++]) / 100.0);
            else total += i;
        }
        return total;
    }
};